### PR TITLE
Cherry-pick #11676 to 6.7: check for a valid limit before we process a memory event

### PR DESCRIPTION
**Pull request by fearful-symmetry**
A backport of #11676.

### CODE REVIEW

File: CHANGELOG.next.asciidoc [hunk 1]
```diff
@@ -51,6 +51,9 @@ https://github.com/elastic/beats/compare/v6.7.0...6.x[Check the HEAD diff]
 
 *Metricbeat*
 
+- Add _bucket to histogram metrics in Prometheus Collector {pull}11578[11578]
```
Comment (kaiyan-sheng) [inline CHANGELOG.next.asciidoc]: Hmm that's a rebase error I guess.

Comment (fearful-symmetry) [reply CHANGELOG.next.asciidoc]: Bah, it's in the other one too.

**Comment by fearful-symmetry** [comment]
jenkins, test this

**Comment by ruflin** [comment]
I think it's because this backport is not merged yet: https://github.com/elastic/beats/pull/11542

**Pull request merged**